src/main/kotlin/com/exactpro/th2/codec/api/impl/ThreadSafeCodec.kt

```diff
@@ -48,13 +52,15 @@ class ThreadSafeCodec(
 
     private fun getInstance() = instances.computeIfAbsent(Thread.currentThread()) {
```

Comment:
```suggestion
    private fun getInstance() = instances.computeIfAbsent(Thread.currentThread().id) {
```
Maybe use id instead of Thread object